plugins/kotlin/uast/uast-kotlin-fir/src/org/jetbrains/uast/kotlin/internal/firKotlinInternalUastUtils.kt

```diff
@@ -42,6 +43,27 @@ internal fun KtAnalysisSession.toPsiMethod(ktCall: KtCall): PsiMethod? {
     }
 }
 
+internal fun KtAnalysisSession.toPsiType(ktType: KtType, context: KtElement): PsiType {
+    if (ktType is KtNonErrorClassType && ktType.typeArguments.isEmpty()) {
+        // TODO: unclear when we need boxed type.
+        val psiType = when (ktType.classId.asFqNameString()) {
+            "kotlin.Int" -> PsiType.INT
+            "kotlin.Long" -> PsiType.LONG
+            "kotlin.Short" -> PsiType.SHORT
+            "kotlin.Boolean" -> PsiType.BOOLEAN
+            "kotlin.Byte" -> PsiType.BYTE
+            "kotlin.Char" -> PsiType.CHAR
+            "kotlin.Double" -> PsiType.DOUBLE
+            "kotlin.Float" -> PsiType.FLOAT
+            "kotlin.Unit" -> PsiType.VOID
```

Comment:
Maybe let's use `org.jetbrains.kotlin.name.StandardClassIds` instead of hardcoding class ids?

Reply:
Good point. Changed both UAST plugins.